### PR TITLE
[EventDispatcher] Remove LegacyEventDispatcherProxy

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Component/EventDispatcher/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+6.0
+---
+
+* Remove `LegacyEventDispatcherProxy`
```
Comment (jderusse) [inline src/Symfony/Component/EventDispatcher/CHANGELOG.md]: Shall we use a short sentence like this in code removal? Or duplicate the message in `UPGRADE-6.0.md` that also contains the `Use the ... instead`?

**Review by jderusse — APPROVED** [review]

File: src/Symfony/Component/EventDispatcher/CHANGELOG.md (hunk 1 above)
Comment (derrabus) [reply src/Symfony/Component/EventDispatcher/CHANGELOG.md]: In this particular case, I don't think that anything beyond the note about the removal is helpful. That class is just plain obsolete now.

**Comment by nicolas-grekas** [comment]
Maybe worth fixing the psalm report in this PR?

File: src/Symfony/Component/EventDispatcher/CHANGELOG.md (hunk 1 above)
Comment (chalasr) [reply src/Symfony/Component/EventDispatcher/CHANGELOG.md]: When we have one, mentioning the replacement in the removal' UPGRADE instruction is good yes

**Comment by derrabus** [comment]
> Maybe worth fixing the psalm report in this PR?

That's a good catch from Psalm. Since EventDispatcher has been bumped to 5.4 everywhere, we can safely drop the calls to `LegacyEventDispatcherProxy` now. I just did that.

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Pull request merged**